purchase_update_seller_price/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2018 JARSA Sistemas S.A. de C.V.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def _add_supplier_to_product(self):
+        """Update the partner info in the supplier list of the product if the
+        supplier is registered for this product."""
+        super(PurchaseOrder, self)._add_supplier_to_product()
+        for line in self.order_line:
+            partner = (
+                self.partner_id if not self.partner_id.parent_id
+                else self.partner_id.parent_id)
+            sellers = line.product_id.seller_ids
+            if partner in sellers.mapped('name'):
+                currency = (
+                    partner.property_purchase_currency_id or
+                    self.env.user.company_id.currency_id)
+                sellers.filtered(lambda x: x.name == partner).write({
```

Comment:
What do you think about saving the price in the currency that you buy? Instead on converting to the currency of the company or the supplier? If you set multi-currency  you can add a price to every currency.

Reply:
This code was taken from the original Odoo method, do you need me to develop it in that way?

Regards

Reply:
It's the same as in the core, so the price computation looks alright to me. Do you also want to write the currency in case they are not the same? As in https://github.com/odoo/odoo/blob/11.0/addons/purchase/models/purchase.py#L454

Reply:
When selecting the supplierinfo to update, do you want to use _select_seller from https://github.com/odoo/odoo/blob/11.0/addons/product/models/product.py#L468, because it looks at minimum quantity and timesliced entries.